homeassistant/components/ovo_energy/sensor.py

```diff
@@ -37,7 +36,39 @@
 class OVOEnergySensorEntityDescription(SensorEntityDescription):
     """Class describing System Bridge sensor entities."""
 
-    value: Callable[[OVODailyUsage], StateType | datetime] = round
+    value: Callable = round
+
+
+def get_electricity_unit_rate_amount(
+    data: OVOCoordinatorData,
+    index: int | None,
+) -> float | None:
+    """Get the value of the rate."""
+    if (
+        index is None
+        or data.plan is None
+        or data.plan.electricity is None
+        or len(data.plan.electricity.unit_rates) < 1
+        or index >= len(data.plan.electricity.unit_rates)
```

Comment:
Maybe it would be better to do the data validation in the backend lib? It would greatly simplify the code when no none checks would be needed inside the integration.